tests/conftest.py

```diff
@@ -391,17 +393,29 @@ def results_dir(request):
     """
     Fixture yielding the path to a directory into which the test can dump its results
 
-    Directories are unique per test, and named after the test name. Everything the tests puts
-    into its directory will to be uploaded to S3. Directory will be placed inside defs.TEST_RESULTS_DIR.
+    Directories are unique per test, and their names include test name and test parameters.
+    Everything the tests puts into its directory will to be uploaded to S3.
+    Directory will be placed inside defs.TEST_RESULTS_DIR.
 
     For example
     ```py
-    def test_my_file(results_dir):
+    @pytest.mark.parametrize("p", ["a", "b"])
+    def test_my_file(p, results_dir):
         (results_dir / "output.txt").write_text("Hello World")
     ```
-    will result in `defs.TEST_RESULTS_DIR`/test_my_file/output.txt.
+    will result in:
+    - `defs.TEST_RESULTS_DIR`/test_my_file/test_my_file[a]/output.txt.
+    - `defs.TEST_RESULTS_DIR`/test_my_file/test_my_file[b]/output.txt.
+
+    When this fixture is called with DoctestItem as a request.node
+    during doc tests, it will return None.
     """
-    results_dir = defs.TEST_RESULTS_DIR / request.node.originalname
+    try:
+        results_dir = (
+            defs.TEST_RESULTS_DIR / request.node.originalname / request.node.name
```

Comment:
Have you tried it with the [cross-snapshot restore test](https://github.com/firecracker-microvm/firecracker/blob/main/tests/integration_tests/functional/test_snapshot_restore_cross_kernel.py#L73-L84)? does it need some changes?

Reply:
I tried to run this one, but the first phase fails with `find: ‘test_results/test_snapshot_phase1’: No such file or directory` which is quite strange since this dir does exist. If I run it locally, it all works just fine. Any ideas why?

Reply:
ok, I fixed the issue. It was because now the `test_results` are compressed by default, so needed to add an option to not compress the dir for snapshot tests.